### ISSUE TITLE
微调的各种问题

### ISSUE THREAD

**Issue by Roych13** [THUDM/ChatGLM3#666]
### Discussed in https://github.com/THUDM/ChatGLM3/discussions/253

<div type='discussions-op-text'>

<sup>Originally posted by **zRzRzRzRzRzRzR** November 10, 2023</sup>

lora微调卡在（pbd）这里，大家有没有遇过？
里面有这个警告，不知道是不是相关，但没找到处理方案：
WARNING - accelerate.utils.other - Detected kernel version 5.4.0, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.

操作系统：Ubuntu 20.04
显卡：A100
python版本： 3.10.13
transfomers： 4.33.0
torch：            2.1.0
微调关键参数如下：
```
LR=1e-4
# NUM_GPUS=4
NUM_GPUS=1
LORA_RANK=8
LORA_ALPHA=32
LORA_DROUPOUT=0.1

# MAX_SOURCE_LEN=512
MAX_SOURCE_LEN=1024
MAX_TARGET_LEN=128
DEV_BATCH_SIZE=1
# GRAD_ACCUMULARION_STEPS=1
GRAD_ACCUMULARION_STEPS=16
MAX_STEP=500
SAVE_INTERVAL=500
```

目前跑全量微调/pt微调（finetune_chatmodel_demo）的两个例子都成功了。但跑finetune_basemodel_demo卡住。

![卡](https://github.com/THUDM/ChatGLM3/assets/39145668/131d1919-ea07-4970-a209-59d5b46c13bf)

底下是python包版本，供参考。
</div>

> accelerate                    0.24.1
> aiofiles                      23.2.1
> aiohttp                       3.8.6
> aiosignal                     1.3.1
> aiowebsocket                  1.0.0.dev2
> altair                        5.1.2
> annotated-types               0.6.0
> antlr4-python3-runtime        4.9.3
> anyio                         3.7.1
> appdirs                       1.4.4
> astor                         0.8.1
> asttokens                     2.4.1
> astunparse                    1.6.3
> async-timeout                 4.0.3
> attrs                         23.1.0
> backoff                       2.2.1
> beautifulsoup4                4.12.2
> blessed                       1.20.0
> blinker                       1.7.0
> blis                          0.7.11
> Brotli                        1.1.0
> cachetools                    5.3.2
> catalogue                     2.0.10
> certifi                       2023.7.22
> cffi                          1.16.0
> chardet                       5.2.0
> charset-normalizer            3.3.2
> chatglm-llm                   1.5.9
> click                         8.1.7
> cloudpathlib                  0.16.0
> colorama                      0.4.6
> coloredlogs                   15.0.1
> comm                          0.2.0
> confection                    0.1.4
> contourpy                     1.2.0
> cpm-kernels                   1.0.11
> cryptography                  41.0.5
> cycler                        0.12.1
> cymem                         2.0.8
> dataclasses-json              0.6.2
> datasets                      2.0.0
> debugpy                       1.8.0
> decorator                     5.1.1
> deepspeed                     0.12.5
> dill                          0.3.7
> distro                        1.8.0
> effdet                        0.4.1
> einops                        0.7.0
> emoji                         2.8.0
> et-xmlfile                    1.1.0
> exceptiongroup                1.1.3
> executing                     2.0.1
> faiss-cpu                     1.7.4
> fastapi                       0.104.1
> ffmpy                         0.3.1
> filelock                      3.13.1
> filetype                      1.2.0
> flatbuffers                   23.5.26
> fonttools                     4.44.1
> frozenlist                    1.4.0
> fschat                        0.2.32
> fsspec                        2023.10.0
> git-python                    1.0.3
> gitdb                         4.0.11
> GitPython                     3.1.40
> gptcache                      0.1.42
> gradio_client                 0.7.0
> greenlet                      3.0.1
> h11                           0.14.0
> h2                            4.1.0
> hjson                         3.1.0
> hpack                         4.0.0
> html2image                    2.0.4.3
> httpcore                      1.0.2
> httptools                     0.6.1
> httpx                         0.25.1
> huggingface-hub               0.17.3
> humanfriendly                 10.0
> hyperframe                    6.0.1
> idna                          3.4
> importlib-metadata            6.8.0
> importlib-resources           6.1.1
> iniconfig                     2.0.0
> inquirer                      3.1.4
> iopath                        0.1.10
> ipykernel                     6.26.0
> ipython                       8.17.2
> jedi                          0.19.1
> jieba                         0.42.1
> Jinja2                        3.1.2
> joblib                        1.3.2
> jsonpatch                     1.33
> jsonpointer                   2.4
> jsonschema                    4.19.2
> jsonschema-specifications     2023.7.1
> jupyter_client                8.6.0
> jupyter_core                  5.5.0
> kiwisolver                    1.4.5
> langchain                     0.0.335
> langchain-experimental        0.0.40
> langcodes                     3.3.0
> langdetect                    1.0.9
> langsmith                     0.0.64
> latex2mathml                  3.76.0
> layoutparser                  0.3.4
> litellm                       0.13.2
> loguru                        0.7.2
> lxml                          4.9.3
> Markdown                      3.5.1
> markdown-it-py                3.0.0
> markdown2                     2.4.10
> markdownify                   0.11.6
> MarkupSafe                    2.1.3
> marshmallow                   3.20.1
> matplotlib                    3.8.2
> matplotlib-inline             0.1.6
> mdtex2html                    1.2.0
> mdurl                         0.1.2
> mpmath                        1.3.0
> msg-parser                    1.2.0
> msgpack                       1.0.7
> multidict                     6.0.4
> multiprocess                  0.70.15
> murmurhash                    1.0.10
> mypy-extensions               1.0.0
> nest-asyncio                  1.5.8
> networkx                      3.2.1
> nh3                           0.2.14
> ninja                         1.11.1.1
> nltk                          3.8.1
> numexpr                       2.8.7
> numpy                         1.24.4
> nvidia-cublas-cu12            12.1.3.1
> nvidia-cuda-cupti-cu12        12.1.105
> nvidia-cuda-nvrtc-cu12        12.1.105
> nvidia-cuda-runtime-cu12      12.1.105
> nvidia-cudnn-cu12             8.9.2.26
> nvidia-cufft-cu12             11.0.2.54
> nvidia-curand-cu12            10.3.2.106
> nvidia-cusolver-cu12          11.4.5.107
> nvidia-cusparse-cu12          12.1.0.106
> nvidia-nccl-cu12              2.18.1
> nvidia-nvjitlink-cu12         12.3.52
> nvidia-nvtx-cu12              12.1.105
> olefile                       0.46
> omegaconf                     2.3.0
> onnx                          1.15.0
> onnxruntime                   1.15.1
> openai                        0.28.1
> opencv-python                 4.8.1.78
> openpyxl                      3.1.2
> orjson                        3.9.10
> outcome                       1.3.0.post0
> packaging                     23.2
> pandas                        2.0.3
> parso                         0.8.3
> pathlib                       1.0.1
> pdf2image                     1.16.3
> pdfminer.six                  20221105
> pdfplumber                    0.10.3
> peft                          0.6.2
> pexpect                       4.8.0
> Pillow                        9.5.0
> pip                           23.3
> platformdirs                  4.0.0
> pluggy                        1.3.0
> portalocker                   2.8.2
> preshed                       3.0.9
> prettytable                   3.9.0
> prompt-toolkit                3.0.40
> protobuf                      4.25.0
> psutil                        5.9.6
> ptyprocess                    0.7.0
> pure-eval                     0.2.2
> py-cpuinfo                    9.0.0
> pyarrow                       14.0.1
> pyarrow-hotfix                0.5
> pyclipper                     1.3.0.post5
> pycocotools                   2.0.7
> pycparser                     2.21
> pydantic                      2.5.2
> pydantic_core                 2.14.5
> pydeck                        0.8.1b0
> pydub                         0.25.1
> pyecharts                     2.0.4
> Pygments                      2.16.1
> PyMuPDF                       1.23.6
> PyMuPDFb                      1.23.6
> pynvml                        11.5.0
> pypandoc                      1.12
> pyparsing                     3.1.1
> pypdf                         3.17.1
> pypdfium2                     4.24.0
> PySocks                       1.7.1
> pytesseract                   0.3.10
> pytest                        7.4.3
> python-dateutil               2.8.2
> python-decouple               3.8
> python-docx                   1.1.0
> python-dotenv                 1.0.0
> python-editor                 1.0.4
> python-iso639                 2023.6.15
> python-magic                  0.4.27
> python-multipart              0.0.6
> python-pptx                   0.6.23
> pytz                          2023.3.post1
> PyYAML                        6.0.1
> pyzmq                         25.1.1
> rapidfuzz                     3.5.2
> rapidocr-onnxruntime          1.3.8
> ray                           2.8.0
> readchar                      4.0.5
> readline                      6.2.4.1
> referencing                   0.30.2
> regex                         2023.10.3
> requests                      2.31.0
> responses                     0.18.0
> rich                          13.6.0
> rpds-py                       0.12.0
> safetensors                   0.4.0
> scikit-learn                  1.3.2
> scipy                         1.11.3
> seaborn                       0.13.0
> selenium                      4.15.2
> semantic-version              2.10.0
> sentence-transformers         2.2.2
> sentencepiece                 0.1.99
> setuptools                    68.0.0
> shapely                       2.0.2
> shellingham                   1.5.4
> shortuuid                     1.0.11
> simplejson                    3.19.2
> six                           1.16.0
> smart-open                    6.4.0
> smmap                         5.0.1
> sniffio                       1.3.0
> socksio                       1.0.0
> sortedcontainers              2.4.0
> soupsieve                     2.5
> spacy                         3.7.2
> spacy-legacy                  3.0.12
> spacy-loggers                 1.0.5
> SQLAlchemy                    2.0.19
> srsly                         2.4.8
> stack-data                    0.6.3
> starlette                     0.27.0
> streamlit                     1.27.2
> streamlit-aggrid              0.3.4.post3
> streamlit-antd-components     0.2.3
> streamlit-chat                0.1.1
> streamlit-chatbox             1.1.11
> streamlit-feedback            0.1.3
> streamlit-option-menu         0.3.6
> strsimpy                      0.2.1
> svgwrite                      1.4.3
> sympy                         1.12
> tabulate                      0.9.0
> tenacity                      8.2.3
> termcolor                     2.3.0
> text2vec                      1.2.9
> textrank4zh                   0.3
> thinc                         8.2.1
> threadpoolctl                 3.2.0
> tiktoken                      0.4.0
> timm                          0.9.12
> tokenizers                    0.13.3
> tokentrim                     0.1.13
> toml                          0.10.2
> tomli                         2.0.1
> tomlkit                       0.12.0
> toolz                         0.12.0
> torch                         2.1.0
> torchaudio                    2.1.0
> torchvision                   0.16.0
> tornado                       6.3.3
> tqdm                          4.66.1
> traitlets                     5.13.0
> transformers                  4.33.0
> transformers-stream-generator 0.0.4
> trio                          0.23.1
> trio-websocket                0.11.1
> triton                        2.1.0
> typer                         0.9.0
> typing_extensions             4.8.0
> typing-inspect                0.9.0
> tzdata                        2023.3
> tzlocal                       5.2
> unstructured                  0.10.30
> unstructured-inference        0.7.11
> unstructured.pytesseract      0.3.12
> urllib3                       2.1.0
> uvicorn                       0.23.2
> uvloop                        0.19.0
> validators                    0.22.0
> wasabi                        1.1.2
> watchdog                      3.0.0
> watchfiles                    0.21.0
> wavedrom                      2.0.3.post3
> wcwidth                       0.2.10
> weasel                        0.3.4
> websocket-client              1.6.4
> websockets                    11.0.3
> wget                          3.2
> wheel                         0.41.2
> wsproto                       1.2.0
> xformers                      0.0.22.post7
> xlrd                          2.0.1
> XlsxWriter                    3.1.9
> xxhash                        3.4.1
> yarl                          1.9.2
> yaspin                        3.0.1
> zipp                          3.17.0

**Comment by zRzRzRzRzRzRzR** [comment]
尝试   trainer.train(resume_from_checkpoint=checkpoint) 
修改为   trainer.train()

**Comment by mengdexing** [comment]
pbd，你输入return就继续往下执行

**Comment by Roych13** [comment]
> 尝试 trainer.train(resume_from_checkpoint=checkpoint) 修改为 trainer.train()

注释掉breakpoint()后可以继续运行了。具体原因不明，注释掉会导致什么后果，暂时也不知道。

**Comment by hgsw** [comment]
也可以根据提示将protobuf的版本暂时降下来，pip install protobuf==3.20.3
